lib/getUniqueId.js

```diff
@@ -0,0 +1,54 @@
+'use strict';
+var Cesium = require('cesium');
+var defined = Cesium.defined;
+
+module.exports = getUniqueId;
+
+// Given a prefix for a new ID, checks the gltf for matching prefixes in top-level objects with IDs and returns a unique ID.
+function getUniqueId(gltf, prefix) {
+    var prefixMatches = [];
```

Comment:
Use a dictionary here instead of an array. Then you can just do lookups below instead of searching the array every time.